Spigot-Server-Patches/0694-Add-fast-alternative-constructor-for-Vector3f.patch

```diff
@@ -0,0 +1,30 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Irmo van den Berge <irmo.vandenberge@ziggo.nl>
+Date: Wed, 10 Mar 2021 21:26:31 +0100
+Subject: [PATCH] Add fast alternative constructor for Vector3f
+
+Signed-off-by: Irmo van den Berge <irmo.vandenberge@ziggo.nl>
+
+diff --git a/src/main/java/net/minecraft/server/Vector3f.java b/src/main/java/net/minecraft/server/Vector3f.java
+index fa25e5d1530e099f73355c2779e0d57a51ad4b6e..65f2aecc4d03c9fd58bb8f6b5464aece93f251bc 100644
+--- a/src/main/java/net/minecraft/server/Vector3f.java
++++ b/src/main/java/net/minecraft/server/Vector3f.java
+@@ -16,6 +16,18 @@ public class Vector3f {
+         this(nbttaglist.i(0), nbttaglist.i(1), nbttaglist.i(2));
+     }
+ 
++    // Paper start - faster alternative constructor
++    private Vector3f(float x, float y, float z, boolean dummy_var) {
++        this.x = x;
++        this.y = y;
++        this.z = z;
++    }
++
++    public static Vector3f createFast(float x, float y, float z) {
```

Comment:
I'd prefer a different name here.
```suggestion
+    public static Vector3f createWithoutValidityChecks(float x, float y, float z) {
```

Reply:
Sounds good to me, Ill change it. I wasnt sure what to name it myself. Any recommendations for the dummy var? Chose boolean because google didnt really give guidelines whats nice to use.

Reply:
You could use `Void` for the dummy value, and pass `null`.

Reply:
I guess the overhead is optimized out anyway, and yeah its better to use a non-type like that. Ill make that change :)